src/Host/Broker/Impl/Sessions/SessionsController.cs

```diff
@@ -54,15 +56,20 @@ public class SessionsController : Controller {
                 interp = _interpManager.Interpreters.First();
             }
 
-            var session = _sessionManager.CreateSession(User.Identity, id, interp, securePassword, profilePath, request.CommandLineArguments);
-            return Task.FromResult<IActionResult>(new ObjectResult(session.Info));
+            try {
+                var session = _sessionManager.CreateSession(User.Identity, id, interp, securePassword, profilePath, request.CommandLineArguments);
+                return Task.FromResult<IActionResult>(new ObjectResult(session.Info));
+            } catch (Exception ex) {
+                return Task.FromResult<IActionResult>(new ApiErrorResult(HttpContext.Response, BrokerApiError.UnableToStartRHost, ex));
+            }
         }
 
         [HttpGet("{id}/pipe")]
         public IActionResult GetPipe(string id) {
             var session = _sessionManager.GetSession(User.Identity, id);
-            if (session?.Process?.HasExited ?? true) {
-                return NotFound();
+            Exception ex = (session?.Process != null && session.Process.HasExited) ? new Win32Exception(session.Process.ExitCode) : null;
```

Comment:
In this cases 404 is not informative. The goal is to give user more information what happened. For example broken installation (comes as 'module is missing'. This looks more useful than 'host start failed, http error 404'.